extern/ttconv/pprdrv_tt.cpp

```diff
@@ -107,6 +107,61 @@ Fixed getFixed(BYTE *s)
     return val;
 } /* end of getFixed() */
 
+/*
+** Get a 16 bit fixed point (2.14) number.
+**
+** The value range is from -2.0 to a little under 2.0.  The two
+** most-significant bits encode the whole part in two's-complement
+** format and the remaining 14 bits encode the numerator n of the
+** fraction n/16384 to be added to the whole part.
+**
+** https://docs.microsoft.com/en-us/typography/opentype/otspec181/otff#data-types
+*/
+F2DOT14 getF2DOT14(BYTE *s)
+{
+    F2DOT14 val={0, 0};
+
+    int sign = s[0] & 0x80;
+    int bit = (s[0] & 0x40) >> 6;
+    if (sign) {
+      val.whole = bit-2;
+    } else {
+      val.whole = bit;
+    }
+    val.fraction = ((s[0] & 0x3f) << 8) + s[1];
+
+    return val;
+}
+
+/*
+** Convert a fixed-point 2.14 number to a newly-allocated string.
+** Use at most six decimals but remove trailing zeroes and possibly
+** the decimal point.
+*/
+char* F2DOT14value(F2DOT14 f)
+{
+    const size_t maxlen = strlen("-1.234567") + 1;
+    char *value = (char*)calloc(sizeof(char), maxlen);
```

Comment:
One might say this is backwards:
```suggestion
    char *value = (char*)calloc(maxlen, sizeof(char));
```